### ISSUE TITLE
PEAR/PSR2 FunctionCallSignature sniffs suggest wrong indent when there are multiple arguments on a line

### ISSUE THREAD

**Issue by bskendig** [squizlabs/PHP_CodeSniffer#820]
phpcs 2.4 (and 2.5) are flagging indentation which I believe to be correct. It seems to want no more than one level of indentation, which would make the code a challenge to read. phpcs 2.3.4 worked correctly.

```
$ cat test.php
<?php
myFunction(
    'foo', (object) array(
        'bar' => function ($x) {
            return true;
        },
        'baz' => false
    )
);
$qux = array_filter(
    $quux, function ($x) {
        return $x;
    }
);

$ phpcs234 test.php

FILE: test.php
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 2 | ERROR | Missing file doc comment
----------------------------------------------------------------------

Time: 24ms; Memory: 3.75Mb

$ phpcs24 test.php

FILE: test.php
----------------------------------------------------------------------
FOUND 6 ERRORS AFFECTING 6 LINES
----------------------------------------------------------------------
  2 | ERROR | [ ] Missing file doc comment
  4 | ERROR | [x] Multi-line function call not indented correctly;
    |       |     expected 4 spaces but found 8
  5 | ERROR | [x] Multi-line function call not indented correctly;
    |       |     expected 4 spaces but found 12
  6 | ERROR | [x] Multi-line function call not indented correctly;
    |       |     expected 4 spaces but found 8
  7 | ERROR | [x] Multi-line function call not indented correctly;
    |       |     expected 4 spaces but found 8
 12 | ERROR | [x] Multi-line function call not indented correctly;
    |       |     expected 4 spaces but found 8
----------------------------------------------------------------------
PHPCBF CAN FIX THE 5 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 20ms; Memory: 3.75Mb
```

**Comment by gsherwood** [comment]
Thanks for reporting this. I've fixed it up now.

It was caused by the sniff getting confused by having multiple arguments on the same line. When it found this, it didn't properly detect that it was inside a new argument and then fell back to checking the indent of each line in the multi-line call.

**Comment by yguedidi** [comment]
The following code:
```php
some_function($arg, array(
    'a' => true,
), array(
    'b' => 'c',
));
```
Break with `Multi-line function call not indented correctly; expected 4 spaces but found 0` on the wrong line (`), array(`).
I think the detected function closing parenthesis is not the right one.

**Comment by gsherwood** [comment]
> I think the detected function closing parenthesis is not the right one.

Looks fine to me. The sniff wants every line inside the function indented.

The PEAR standard actually wants the code like this (you can indent the array contents more if you want):
```php
some_function(
    $arg, array(
    'a' => true,
    ), array(
    'b' => 'c',
    )
);
```

**Comment by yguedidi** [comment]
@gsherwood thank you for you're answer, but PSR-2 don't, have a look to [PSR-2 Errata](http://www.php-fig.org/psr/psr-2/meta/#3-1-multi-line-arguments-09-08-2013)

**Comment by gsherwood** [comment]
@yguedidi I thought you were talking about the PEAR standard because the PSR2 standard doesn't produce any errors on that code. If you are getting errors, make sure you are using the latest PHPCS version (2.6.1).

**Comment by yguedidi** [comment]
@gsherwood My bad, it was my custom standard that ref to PSR2 but also to Zend, so my standard has both PSR2 and PEAR version of `FunctionCallSignature`, Zend one include it explicitly...
Excluding the PEAR one from my standard make the output green!
Thank you for you help and time!

**Comment by gsherwood** [comment]
@yguedidi No problem. Glad you figured it out.
